src/blocks/HeaderBlockBrandPivot/index.tsx

```diff
@@ -0,0 +1,366 @@
+    },
+  }),
+)
+const Filler = styled('div')({
+  height: WRAPPER_HEIGHT,
+})
+const HeaderBackgroundFiller = styled('div')<{ transparent: boolean }>(
+  ({ transparent }) => ({
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    right: 0,
+    zIndex: -1,
+    height: WRAPPER_HEIGHT,
+    backgroundColor: colorsV3.white,
+    opacity: transparent ? 0 : 1,
+    transition: 'opacity 300ms',
+  }),
+)
+
+const InnerHeaderWrapper = styled('div')({
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  width: '100%',
+  height: WRAPPER_HEIGHT,
+  padding: HEADER_VERTICAL_PADDING + ' 0',
+})
+const Menu = styled('ul')<{ open: boolean }>(({ open }) => ({
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  margin: 0,
+  padding: 0,
+  listStyle: 'none',
+
+  [TABLET_BP_DOWN]: {
+    position: 'absolute',
+    display: 'block',
+    zIndex: 101,
+    width: '80%',
+    height: '100vh',
+    top: 0,
+    bottom: 0,
+    right: open ? '20%' : '100%',
+    paddingTop: `calc(${WRAPPER_HEIGHT} + ${HEADER_VERTICAL_PADDING})`,
+    fontSize: 18,
+    background: colorsV3.white,
+    transition: `right ${TOGGLE_TRANSITION_TIME}ms`,
+    color: colorsV3.gray900,
+    overflow: 'scroll',
+    WebkitOverflowScrolling: 'touch',
+  },
+}))
+
+const LogoLink = styled('a')({
+  display: 'inline-flex',
+  paddingTop: 3, // fix to push down logo a little or it looks unbalanced
+  color: 'inherit',
+  marginLeft: '1rem',
+})
+
+const ButtonWrapper = styled('div')({
+  paddingLeft: '2rem',
+
+  [TABLET_BP_DOWN]: {
+    paddingTop: '1.5rem',
+    paddingLeft: '1rem',
+    paddingBottom: HEADER_VERTICAL_PADDING,
+  },
+})
+
+const RightContainer = styled('div')({
+  display: 'flex',
+})
+
+const MobileHeaderLink = styled(ButtonLinkBrandPivot)({
+  [TABLET_BP_UP]: {
+    display: 'none',
+  },
+})
+
+interface HeaderBlockProps extends BaseBlockProps {
+  is_transparent: boolean
+  inverse_colors: boolean
+  override_cta_link?: LinkComponent | null
+  override_cta_label?: string | null
+  cta_color?: MinimalColorComponent
+  cta_style?: ButtonStyleType
+  override_mobile_header_cta_label?: string | null
+  override_mobile_header_cta_link?: LinkComponent | null
+  mobile_header_cta_color?: MinimalColorComponent
+  mobile_header_cta_style?: ButtonStyleType
+}
+
+enum InverseColors {
+  DEFAULT = 'standard-inverse',
+  INVERSE = 'standard',
+}
+
+export const Header: React.FC<{ story: GlobalStory } & HeaderBlockProps> = (
+  props,
+) => {
+  const [isBelowThreshold, setScrollPosition] = React.useState<boolean>(
+    isBelowScrollThreshold(),
+  )
+  const [buttonColor, setButtonColor] = React.useState<
+    minimalColorComponentColors | undefined
+  >(props.cta_color?.color)
+
+  React.useEffect(() => {
+    if (!props.is_transparent) {
+      return
+    }
+
+    setTimeout(() => {
+      updateHeader() // update the scroll animation after mount because the server doesn't know scroll position
```

Comment:
isn't `useEffect` triggered after mount already? if it isn't then `setTimeout` is a bit of a tankevurpa anyways i believe because `setTimeout` works in node also. If we want something on the client side only we should probably check for `typeof window !== 'undefined'` or something simliar, but considering we're using `window.addEventListener` below i'm quite sure this does not run on the server. Maybe we should run this on `window.onload` or something rather? or just not use `setTimeout` in the first place. Or at least remove this comment since it's not true ;)

Reply:
Yes you're absolutely right, the timeout should be removed as well. Made sure we set `isBelowThreshold` state to false so we do the check in the hook instead and that fixed the issue when you're reload and are already scrolled down on the page.